stubs/jsonschema/jsonschema/protocols.pyi

```diff
@@ -3,27 +3,31 @@ from collections.abc import Iterator, Mapping, Sequence
 from typing import ClassVar, Protocol
 from typing_extensions import TypeAlias
 
+from jsonschema import _typing
 from jsonschema._format import FormatChecker
 from jsonschema._types import TypeChecker
 from jsonschema.exceptions import ValidationError
 from jsonschema.validators import RefResolver
+from referencing.jsonschema import SchemaRegistry
 
 _JsonParameter: TypeAlias = str | int | float | bool | None | Mapping[str, _JsonParameter] | Sequence[_JsonParameter]
 
 class Validator(Protocol):
-    META_SCHEMA: ClassVar[dict[Incomplete, Incomplete]]
-    VALIDATORS: ClassVar[dict[Incomplete, Incomplete]]
+    META_SCHEMA: ClassVar[Mapping[Incomplete, Incomplete]]
+    VALIDATORS: ClassVar[Mapping[Incomplete, Incomplete]]
     TYPE_CHECKER: ClassVar[TypeChecker]
     FORMAT_CHECKER: ClassVar[FormatChecker]
-    schema: dict[Incomplete, Incomplete] | bool
+    ID_OF: _typing.id_of
+    schema: Mapping[Incomplete, Incomplete] | bool
     def __init__(
         self,
-        schema: dict[Incomplete, Incomplete] | bool,
+        schema: Mapping[Incomplete, Incomplete] | bool,
+        registry: SchemaRegistry,
         resolver: RefResolver | None = None,
```

Comment:
This parameter doesn't seem to exist in the latest version of jsonschema.

Reply:
That's true. Here's an excerpt from the PR description above regarding this decision:
> I left in the resolver param which was removed from the library's signature in the same release, to be replaced by registry. However, according to the docs, resolver is only deprecated and it is still entirely possible to use it. Removing it from the stub would lead to the same error for all current users of the resolver param. This goes somewhat counter to [CONTRIBUTING.md](https://github.com/python/typeshed/blob/main/CONTRIBUTING.md#what-to-do-when-a-projects-documentation-and-implementation-disagree) but I felt this scenario is a little different than what is described there. This looks like an "error" in the function signature, not the docs.

I really wasn't sure how to handle this case. Please let me know if I should remove the param.

Reply:
Normally I'd say we follow the implementation, but the fact that this is a Protocol complicates things a bit. When you say "it is still entirely possible to use it" do you mean concrete implications of the protocol do accept this parameter?

Also, should these parameters perhaps be marked positional-only?

Reply:
> Normally I'd say we follow the implementation, but the fact that this is a Protocol complicates things a bit. When you say "it is still entirely possible to use it" do you mean concrete implications of the protocol do accept this parameter?

Yes, that is what I mean. For reference, here's the docs on the *Validator protocol* still mentioning the `resolver` param: 
https://github.com/python-jsonschema/jsonschema/blob/08395e9655f680874c55fbba5f95c36aacc90849/jsonschema/protocols.py#L57-L65

And here's where the *Validator class* still declares the `_resolver` attribute - thereby ending in a `resolver` param being included in its `__init__()` through *attrs* magic: 
https://github.com/python-jsonschema/jsonschema/blob/08395e9655f680874c55fbba5f95c36aacc90849/jsonschema/validators.py#L244-L249

> Also, should these parameters perhaps be marked positional-only?

I might be missing something, because I'm not super proficient in any of this. Why do you suggest that? Neither the protocol nor the implementation declare these positional-only.

Reply:
Sorry, I meant keyword-only not positional-only. I was thinking of that because adding or removing a positional parameter in the middle of a list won't work well.

Reply:
I see, that makes sense to me.
Honestly, I'm a bit lost. You're the expert, not me. 
I assume the typeshed guidelines were written specifically to avoid this kind of interpretation of the intentions of library authors. So I probably shouldn't even have started this discussion. Would you be okay if I just removed the `resolver` param?
As for the keyword-only issue, I'm not sure if the guidelines have anything to say on that point but I guess the gist is as always: _follow the implementation_, right?

Reply:
Yes, let's default to following the implementation. If a user reports an issue we can revisit the decision.